MediaManager/Media/MediaItem.cs

```diff
@@ -11,6 +11,15 @@ public MediaItem(string uri)
             if (string.IsNullOrEmpty(uri))
                 throw new ArgumentNullException(uri);
             MediaUri = uri;
+            MediaLocation = GetLocationType();
+        }
+
+        MediaLocation GetLocationType()
```

Comment:
Lets not put this in the MediaItem class. Maybe move it into a new `MediaResolver` class and rename method to `GetMediaLocation()`. Also make virtual then so people can override.

Reply:
Moved to `MediaExtractor`.